adservertargeting/respdataprocessor.go

```diff
@@ -0,0 +1,201 @@
+package adservertargeting
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/prebid/openrtb/v17/openrtb2"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	jsonpatch "gopkg.in/evanphx/json-patch.v4"
+	"strings"
+)
+
+const MaxKeyLength = 20
+
+func processRequestTargetingData(adServerTargetingData *adServerTargetingData, targetingData map[string]string, bidImpId string) {
+	if len(adServerTargetingData.RequestTargetingData) == 0 {
```

Comment:
The length check isn't necessary. Keep in mind that Go's range loop gracefully handles nil or empty lists such that no optimization is necessary.